addons/hr/models/res_users.py

```diff
@@ -1,21 +1,156 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, fields
+from odoo import api, models, fields, _
+from odoo.exceptions import AccessError
 
 
 class User(models.Model):
+
     _inherit = ['res.users']
 
-    employee_ids = fields.One2many('hr.employee', 'user_id', string='Related employees')
+    # note: a user can only be linked to one employee per company
```

Comment:
How to you prevent this use case ? 

Shouldn't we prevent this in the create/write method ?

Reply:
With a  `unique(user_id, company_id)` constraint in `hr.employee`  ([here](https://github.com/odoo/odoo/blob/6d1dfa1d48bb9c91b0a66882421427ec38d6dcc8/addons/hr/models/hr.py#L215))